### DISCUSSION TITLE
Case key paths for receiving test store actions cannot be compiled.

### DISCUSSION THREAD

**Discussion by Kyome22** [pointfreeco/swift-composable-architecture#3553] [General]
### Description

If the `enum` defined with associated values and its value type is `Result` with a success value of tuple, compiling a receiving action using case key paths will fail.
(Even if I implement it as written in Migrating 1.4, the build will fail, and the cause will not be clearly displayed in the log.)

### Checklist

- [X] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [ ] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-composable-architecture/issues) or [discussion](https://github.com/pointfreeco/swift-composable-architecture/discussions).

### Expected behavior

Compilation successes.

### Actual behavior

Compilation fails.

### Reproducing project

```swift
import ComposableArchitecture
import XCTest

@Reducer
struct SomeReducer {
    @ObservableState
    struct State: Equatable {}

    enum Action {
        case response(Result<(Bool, String), any Error>)
    }

    var body: some ReducerOf<Self> {
        EmptyReducer()
    }
}

final class SomeReducerTests: XCTestCase {
    @MainActor
    func test_sample() async {
        let store = TestStore(initialState: SomeReducer.State()) {
            SomeReducer()
        }
        await store.receive(\.response.failure) // cannot be compiled
    }
}
```

If you use a struct instead of a tuple, it will compile.

### The Composable Architecture version information

1.17.0

### Destination operating system

iOS 18

### Xcode version information

Version 16.2 (16C5032a)

### Swift Compiler version information

```shell
swift-driver version: 1.115.1 Apple Swift version 6.0.3 (swiftlang-6.0.3.1.10 clang-1600.0.30.1)
Target: arm64-apple-macosx14.0
```

**Comment by mbrandonw** [comment]
Hi @Kyome22, this unfortunately seems like a crash in the compiler and not related to TCA. Here is a simpler reproduction:

```swift
@CasePathable enum Action { case result(Result<(Int, Int), Never>) }
func foo<A>(casePath: CaseKeyPath<Action, A>) {}
func bar() {
  foo(casePath: \.result.success)  // 🛑
  foo(casePath: \.result.failure)  // 🛑
}
```

It works fine when the tuple is replaced with a non-tuple. I recommend bundling the data of your tuple up into a proper struct, and then it will work fine.

Since this isn't an issue with the library I am going to convert it to a discussion. Please feel free to continue the conversation over there!

**Comment by stephencelis** [comment]
@Kyome22 If you open an issue on the Swift repo please share here!
